modules/mini-games/team_production/map_scripts.lua

```diff
@@ -0,0 +1,81 @@
+
+local Public = {}
+
+function Public.create_tiles(distance, tiles, offset_x, offset_y, script_data)
+
+  local count = 1
+  local map_tiles = {}
+  local chunks = {}
+
+  local chunk_distance = math.ceil(distance / 32)
+  local chunk_offset_x = math.floor(offset_x / 32)
+  local chunk_offset_y = math.floor(offset_y / 32)
+  for x = -chunk_distance, chunk_distance do
+    for y = -chunk_distance, chunk_distance do
+      table.insert(chunks, {x = x + chunk_offset_x, y = y + chunk_offset_y})
+    end
+  end
+
+  for name, array_x in pairs (tiles) do
+    for X, array_y in pairs (array_x) do
+      for k, Y in pairs (array_y) do
+        local x = X + offset_x
+        local y = Y + offset_y
+        map_tiles[count] = {name = name, position = {x, y}}
+        count = count + 1
+      end
+    end
+  end
+
+  script_data.surface.set_tiles(map_tiles, true)
+  script_data.surface.regenerate_decorative(nil, chunks)
+end
+
+function Public.recreate_entities(entities, offset_x, offset_y, force, duration, script_data)
+  if not entities or not force or not offset_x or not duration or not offset_y then return end
+  local tick = game.tick
+  local surface = script_data.surface
+  for name, array in pairs (entities) do
+    for k, entity in pairs (array) do
+      if (k + tick) % duration == 0 then
+        local position = {entity.position[1] + offset_x, entity.position[2] + offset_y}
+        if entity.amount then
+          surface.create_entity{name = name, position = position, amount = entity.amount}
+        elseif name == "stack-inserter" then
+          local v = surface.create_entity{name = name, position = position, force = force, direction = entity.direction}
+          v.destructible = false
+          v.minable = false
+          v.rotatable = false
+        elseif name == "red-chest" then
+          local v = surface.create_entity({force = force, name = name, position = position})
```

Comment:
v is not the best variable name 😛 i would make it ent for entity

Reply:
Not my code so I didnt change it.

Reply:
fine with me